test/mocha/helpers.js

```diff
@@ -0,0 +1,241 @@
+const brIdentity = require('bedrock-identity');
+const brLdnReceiver = require('bedrock-ldn-receiver');
+const brKey = require('bedrock-key');
+const config = require('bedrock').config;
+const database = require('bedrock-mongodb');
+const jsigs = require('jsonld-signatures');
+const uuid = require('uuid/v4');
+const url = require('url');
+
+const api = {};
+module.exports = api;
+
+api.IDENTITY_BASE_PATH = config.server.baseUri +
+  config['identity-http'].basePath + '/';
+
+api.createCrytographicIdentity = function(sourceIdentity, callback) {
+  var publicKey = {
+    '@context': 'https://w3id.org/identity/v1',
+    id: sourceIdentity.keys.publicKey.id,
+    type: 'CryptographicKey',
+    owner: sourceIdentity.identity.id,
+    publicKeyPem: sourceIdentity.keys.publicKey.publicKeyPem
+  };
+  var credential = {
+    '@context': 'https://w3id.org/identity/v1',
+    id: 'urn:ephemeral:' + uuid(),
+    type: ['Credential', 'CryptographicKeyCredential'],
+    claim: {
+      id: sourceIdentity.identity.id,
+      publicKey: publicKey
+    }
+  };
+  jsigs.sign(
+    credential, {
+      privateKeyPem:
+        sourceIdentity.keys.privateKey.privateKeyPem,
+      creator: sourceIdentity.keys.publicKey.id,
+      // domain: url.parse(config.server.baseUri).host,
+      algorithm: 'LinkedDataSignature2015'
+    }, (err, signedCredential) => {
+      if(err) {
+        callback(err);
+      }
+      var targetIdentity = {
+        '@context': 'https://w3id.org/identity/v1',
+        id: sourceIdentity.identity.id,
+        type: 'Identity',
+        credential: [{'@graph': signedCredential}]
+      };
+      jsigs.sign(
+        targetIdentity, {
+          privateKeyPem:
+            sourceIdentity.keys.privateKey.privateKeyPem,
+          creator: sourceIdentity.keys.publicKey.id,
+          domain: url.parse(config.server.baseUri).host,
+          algorithm: 'LinkedDataSignature2015'
+        }, (err, signedIdentity) => {
+          if(err) {
+            callback(err);
+          }
+          callback(null, signedIdentity);
+        });
+    });
+};
+
+api.createInbox = function(mockData, options = {addId: true}) {
+  const inbox = bedrock.util.clone(mockData.inbox);
+  if(options.addId) {
+    inbox.id = brLdnReceiver.createInboxId();
+  }
+  return inbox;
+};
+
+api.createRawInbox = function(mockData, options = {addId: true}) {
+  const inbox = bedrock.util.clone(mockData.rawInbox);
+  if(options.addId) {
+    inbox.id = brLdnReceiver.createInboxId();
+  }
+  return inbox;
+};
+
+api.createMessage = function(mockData, options = {addId: true}) {
+  const message = bedrock.util.clone(mockData.message);
+  if(options.addId) {
+    message.id = brLdnReceiver.createMessageId();
+  }
+  return message;
+};
+
+api.createRawMessage = function(mockData, options = {addId: true}) {
+  const message = bedrock.util.clone(mockData.rawMessage);
+  if(options.addId) {
+    message.id = brLdnReceiver.createMessageId();
+  }
+  return message;
+};
```

Comment:
Why do we have different methods with the same implementation here except for the input (`mockData.rawFoo` vs. `mockData.foo`)? Shouldn't we just pass in `mockData.rawFoo`/`mockData.foo` directly and have only one function?